packages/genprint/genprint.0.1/opam

```diff
@@ -0,0 +1,35 @@
+opam-version: "2.0"
+#name: "genprint"
+#version: "0.1"
+
+synopsis: "PPX syntax extension and library package for printing values of any type"
+description: """
+A PPX syntax extension and library package enabling printing of values of any type using OCaml's internal printing facilities a la toplevel evaluation.
+Useful for debugging as a quick alternative to ocamldebug/ppx_deriving/#install_printer.
+"""
+
+maintainer: "progman1"
+authors: [ "progman1" ]
+homepage: "https://github.com/progman1/genprintlib"
+bug-reports: "https://github.com/progman1/genprintlib/issues"
+dev-repo: "git+https://github.com/progman1/genprintlib.git"
+
+build: [
+  ["dune" "subst"] {pinned}
+  ["dune" "build"  "@epoc1"] {ocaml:version  <"4.08.0"}
+  ["dune" "build"  "@epoc2"] {ocaml:version >="4.08.0"}
```

Comment:
those two call to dune are only advised if you're developing, not for releases.
I looked at the code and it would be much more simple to just have a rule that generate the `genprintval.ml` from a generator program (in OCaml and thus would have access to the version number)